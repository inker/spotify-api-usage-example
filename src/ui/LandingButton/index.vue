<template>
  <button
    :type="type"
    class="login-button"
    v-bind="$attrs"
    v-on="$listeners"
  >
    <slot />
  </button>
</template>

<script>
export default {
  inheritAttrs: false,

  props: {
    type: {
      type: String,
      default: 'button',
    },
  },
}
</script>

<style lang="postcss" scoped>
.login-button {
  margin: 0;
  width: 12rem;
  height: 3rem;
  cursor: pointer;
  border: none;
  border-radius: 999px;
  background-color: var(--blue);
  color: var(--white);

  &:hover {
    filter: brightness(1.1);
  }
}
</style>
