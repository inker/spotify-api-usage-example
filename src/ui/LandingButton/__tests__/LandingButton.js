import { shallowMount } from '@vue/test-utils'

import LandingButton from '..'

describe('LandingButton.vue', () => {
  test('default type', () => {
    const noPropsButton = shallowMount(LandingButton)
    const buttonWithExplicitType = shallowMount(LandingButton, {
      propsData: {
        type: 'button',
      },
    })
    expect(noPropsButton.html()).toMatchSnapshot(buttonWithExplicitType.html())
  })

  test('submit button', () => {
    const wrapper = shallowMount(LandingButton, {
      propsData: {
        type: 'submit',
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
