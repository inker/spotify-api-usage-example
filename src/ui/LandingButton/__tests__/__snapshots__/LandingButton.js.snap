// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingButton.vue default type: <button type="button" class="login-button"></button> 1`] = `<button type="button" class="login-button"></button>`;

exports[`LandingButton.vue submit button 1`] = `<button type="submit" class="login-button"></button>`;
